### ISSUE TITLE
contentDir/configDir per Lang CLI flag

### ISSUE THREAD

**Issue by onedrawingperday** [gohugoio/hugo#5662]
Following the recent enforcement of the Hugo Basic Example `contentDir` in the Build Script of the Hugo Themes site I discovered that if this parameter is present in the `config` of an Example Site then Hugo picks up what is entered in the `config` and not the CLI flag in the Build Script.

So far I have found one theme with a `contentDir` [parameter](https://github.com/shaform/hugo-theme-den/blob/master/exampleSite/config.toml#L114). As a result its [demo](https://themes.gohugo.io/theme/hugo-theme-den/en) is published using the content that the author wants and not what the Build Script uses.

(Note that even though the above theme uses multilingual features this doesn't matter. I tested with other themes locally and if I enter this parameter in the Example Site `config` of any theme then again it takes precedence over the CLI flag.)

We don't quite want that anymore for the Hugo Themes website, after the recent cases of inappropriate content.

Would it be feasible to have the CLI flag take precedence over the `config`?

cc: @digitalcraftsman

**Comment by bep** [comment]
I started adding a failing test for this, that didn't fail. And then I looked at your example. The thing is:

* --configDir sets the main configDir setting, which works
* But then you can set a configDir per language

I need to think a little about this, but in a theme context, it would not make much sense to use the same content for each language ...

**Comment by onedrawingperday** [comment]
I updated the issue title to reflect the above findings.

To recap: 
- contentDir or configDir CLI refers to the main respective directories.
- contentDir or configDir CLI per lang do not have CLI flags currently.

So basically this issue is an enhancement. 

However note that I do not particularly care about this issue myself. 
I only reported it because I came across it in the Hugo Themes site.
